src/main/java/net/snowflake/client/core/SFStatement.java

```diff
@@ -188,7 +186,7 @@ SFBaseResultSet executeQueryInternal(
       throws SQLException, SFException {
     resetState();
 
-    logger.debug("executeQuery: {}", (ArgSupplier) () -> SecretDetector.maskSecrets(sql));
+    logger.debug("executeQuery: {}", sql);
```

Comment:
Is the `sql` variable already sanitized?

Reply:
No, it will be sanitized in either SLF4JLogger.java or JDK14Logger.java. This line was causing it to be sanitized twice.

void debug(String msg, Object... arguments); -> this is unchanged and always get sanitized 
e.g: https://github.com/snowflakedb/snowflake-jdbc/blob/master/src/main/java/net/snowflake/client/log/SLF4JLogger.java#L72

void debug(String msg, boolean isMasked); -> would depend on isMasked
Previously all messages were always being masked:
https://github.com/snowflakedb/snowflake-jdbc/blob/master/src/main/java/net/snowflake/client/log/SLF4JLogger.java#L52

Reply:
No, it will be sanitized in either SLF4JLogger.java or JDK14Logger.java. This line was causing it to be sanitized twice.

void debug(String msg, Object... arguments); -> this is unchanged and always get sanitized
e.g: https://github.com/snowflakedb/snowflake-jdbc/blob/master/src/main/java/net/snowflake/client/log/SLF4JLogger.java#L72

void debug(String msg, boolean isMasked); -> would depend on isMasked
Previously all messages were always being masked:
https://github.com/snowflakedb/snowflake-jdbc/blob/master/src/main/java/net/snowflake/client/log/SLF4JLogger.java#L52